### DISCUSSION TITLE
Pre-build tool to measure ONNX runtime performance for models?

### DISCUSSION THREAD

**Discussion by Voxelworld** [microsoft/onnxruntime#21050] [Performance Q&A]
I would like to check whether the ONNX-Runtime is capable to execute my model (.onnx file) on CPU and with the DirectML execution provider and would like to capture performance measurements on the GPU and CPU, like memory consumption and execution time.

The official ONNX-Runtime documentation [states](https://onnxruntime.ai/docs/performance/tune-performance/profiling-tools.html) *"The onnxruntime_perf_test.exe tool (available from the build drop) can be used to test various knobs."*. But I can't find the tool in the release assets.

Previously I used tools in Microsofts GitHub repo [Windows-Machine-Learning](https://github.com/microsoft/Windows-Machine-Learning/releases/tag/v1.3.0) which contains a *WinMLRunner.exe* and a *MicrosoftMLRunner.exe*. But the last version was released in 2022.

Where can I find a performance measuring tool which can be used out of the box?

**Comment by AndreyOrb** [comment]
As @snnn pointed out in https://github.com/microsoft/onnxruntime/issues/21378#issuecomment-2231913628, you can use [MLPerf](https://github.com/mlcommons/inference/tree/master/vision/classification_and_detection)
